arraycontext/impl/pyopencl/__init__.py

```diff
@@ -133,6 +156,9 @@ def from_numpy(self, array: np.ndarray):
         return cl_array.to_device(self.queue, array, allocator=self.allocator)
 
     def to_numpy(self, array):
+        if not self._force_device_scalars and np.isscalar(array):
```

Comment:
Why `isscalar` over `isinstance(..., Number)`?

Reply:
It also catches `np.bool_` and `np.bool8`.

https://github.com/inducer/arraycontext/pull/43#discussion_r656733216